### PR TITLE
Disable CTRL Encoder

### PR DESCRIPTION
**Pull request by connor-mccorm**

### CODE REVIEW

**Review by arnavgarg1 — COMMENTED** [review]
Do we need to remove it from the schema?

Also, can we create an issue on Github and link it here as a comment?

**Comment by connor-mccorm** [comment]
@arnavgarg1 So by commenting out this line, it actually removes this encoder from the schema since it won't be in the list of registered encoders when generating the schema. Then as soon as it's functional again, we can just uncomment this line to add it back in.

**Review by tgaddair — APPROVED** [review]

**Comment by tgaddair** [comment]
@connor-mccorm can you create a GH issue with the error we were seeing:

```
ray.exceptions.RayTaskError(RuntimeError): [36mray::tune_batch_size_fn()[39m (pid=95, ip=192.168.52.234)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/ludwig/backend/ray.py", line 253, in tune_batch_size_fn
    return trainer.tune_batch_size(ludwig_config, train_shard, **kwargs)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/ludwig/trainers/trainer.py", line 429, in tune_batch_size
    samples_per_sec = self.train_for_tuning(batch_size, total_steps=5)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/ludwig/trainers/trainer.py", line 377, in train_for_tuning
    self.train_step(inputs, targets)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/ludwig/trainers/trainer.py", line 263, in train_step
    model_outputs = self.dist_model((inputs, targets))
  File "/home/ray/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/ludwig/models/ecd.py", line 140, in forward
    encoder_outputs = self.encode(inputs)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/ludwig/models/ecd.py", line 80, in encode
    encoder_output = encoder(input_values)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/ludwig/features/text_feature.py", line 196, in forward
    encoder_output = self.encoder_obj(inputs_exp, mask=inputs_mask)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/ludwig/encoders/text_encoders.py", line 1310, in forward
    transformer_outputs = self.transformer.module(
  File "/home/ray/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/transformers/models/ctrl/modeling_ctrl.py", line 459, in forward
    pos_embeds = self.pos_encoding[position_ids, :].to(device)
RuntimeError: indices should be either on cpu or on the same device as the indexed tensor (cpu) (type: RayTaskError(RuntimeError), retryable: true)
```

Then link the issue to this PR and add a `TODO` comment above the class definition so we know why it's disabled.

**Review by tgaddair — CHANGES_REQUESTED** [review]
See comment.

**Review by abidwael — APPROVED** [review]

File: ludwig/schema/encoders/text_encoders.py [hunk 1]
```diff
@@ -1906,7 +1906,7 @@ def module_name():
 
 
 @DeveloperAPI
-@register_encoder_config("ctrl", TEXT)
+# @register_encoder_config("ctrl", TEXT)  TODO: uncomment when CTRL bug has been fixed to add back in
```
Comment (arnavgarg1) [inline ludwig/schema/encoders/text_encoders.py:1909]: @connor-mccorm Can you link the issue?

Comment (tgaddair) [reply ludwig/schema/encoders/text_encoders.py:1909]: +1

Comment (connor-mccorm) [reply ludwig/schema/encoders/text_encoders.py:1909]: Will do!

**Review by arnavgarg1 — APPROVED** [review]

**Review by tgaddair — APPROVED** [review]

**Review by justinxzhao — APPROVED** [review]

**Pull request merged**